controllers/wizard/fileUpload/PKPFileUploadWizardHandler.inc.php

```diff
@@ -63,6 +66,116 @@ function __construct() {
 	//
 	// Implement template methods from PKPHandler
 	//
+	function authorize($request, &$args, $roleAssignments) {
```

Comment:
This kind of reiteration of rules also shouldn't live here, but the upload wizard already has that kind of pollution and IMO we'll need a 3rd-party implementation to eventually reconcile all of this into.

Reply:
Agreed. One of the things that I've come to feel is that `fileStage` is maybe not appropriate for review attachments, discussion files, and dependent files. Authorization to read/write these files is done independently of their file stage. I've got some notes down to discuss this once the current work is all done and dusted.